src/main/java/software/sigma/bu003/internship/vedmid_andrii/exception/GlobalControllerExceptionHandler.java

```diff
@@ -0,0 +1,19 @@
+package software.sigma.bu003.internship.vedmid_andrii.exception;
+
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class GlobalControllerExceptionHandler {
+
+    @ExceptionHandler(PartNotFoundException.class)
```

Comment:
add default handling for 500 error too, it;s not an exception - move to controller package, also you need a pretty pojo for handling ErrorInfo etc "return new ErrorInfo(req.getRequestURL().toString(), ex);"